### PR TITLE
[CI] Fix dashboard

### PR DESCRIPTION
**Pull request by architkulkarni**
A new @types/react release has broken the dashboard build. Make sure to specify the older version under package resolutions.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by architkulkarni** [comment]
@alanwguo @edoakes Could you please approve this PR when you get a chance?  Cherry picks require one approval from a TL and one approval from a Ray committer

**Review by edoakes — APPROVED** [review]

**Comment by architkulkarni** [comment]
@alanwguo @clarkzinzow Actually, a lot of dashboard tests failed in this PR, so I don't think it's safe to merge.  Any idea how to fix these?  Example: https://buildkite.com/ray-project/ray-builders-pr/builds/29102#38b85f48-0ad1-4f9d-aaf0-c77782f1e9c8/6-11551

**Comment by clarkzinzow** [comment]
@architkulkarni The failures are unrelated to this change, they're due to `prometheus_client` not being upper-bounded, and latest `prometheus_client` release (0.14.0) breaks Ray 1.11+ (at least). We fixed this in master by upper-bounding `prometheus_client` by 0.13.1, so you should just need to cherry-pick [this commit](https://github.com/ray-project/ray/commit/d27e73f85171ca23ed74175a3cdba6e555c4d540).

**Comment by architkulkarni** [comment]
> @architkulkarni The failures are unrelated to this change, they're due to `prometheus_client` not being upper-bounded, and latest `prometheus_client` release (0.14.0) breaks Ray 1.11+ (at least). We fixed this in master by upper-bounding `prometheus_client` by < 0.14.0, so you should just need to cherry-pick [this commit](https://github.com/ray-project/ray/commit/d27e73f85171ca23ed74175a3cdba6e555c4d540).

Ah of course, thanks, I forgot about that one!

**Review by clarkzinzow — APPROVED** [review]

**Pull request merged**
